### PR TITLE
fix: python sdk can parse master output again [DET-8021]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

The GET experiments endpoint changed, but the python sdk did not.

Follow-up work includes migrating all of the python sdk over to
use the generated bindings, but that was no way going to happen
in time for the 0.19.1 release.

### CODE REVIEW

File: harness/determined/common/experimental/experiment.py [hunk 1]
```diff
@@ -131,7 +130,7 @@ def pause(self) -> None:
     def unarchive(self) -> None:
         self._session.post(f"/api/v1/experiments/{self.id}/unarchive")
 
-    def wait(self, interval: int = 5) -> ExperimentState:
+    def wait(self, interval: float = 0.5) -> ExperimentState:
```
Comment (azhou-determined) [inline harness/determined/common/experimental/experiment.py:133]: what's the reasoning behind this?

Comment (rb-determined-ai) [reply harness/determined/common/experimental/experiment.py:133]: oops, I meant to change it to a `float`, not to change the default value

File: harness/determined/common/experimental/model.py [hunk 2]
```diff
@@ -300,9 +305,14 @@ def set_labels(self, labels: List[str]) -> None:
...
-            labels (List[string]): All labels to set on the model.
+            labels (List[str]): All labels to set on the model.
         """
-        self.labels = labels
+        if not isinstance(labels, Iterable):
+            raise ValueError(
+                f"set_labels() requires a list of strings as input but got: {labels}"
+            )
+
+        self.labels = list(labels)
```
Comment (azhou-determined) [inline harness/determined/common/experimental/model.py:315]: why is this list() call necessary?

Comment (rb-determined-ai) [reply harness/determined/common/experimental/model.py:313]: Previously we'd write whatever you put onto the wire.

I added a check, but not a check to make sure it was actually a `list`, just that we could iterate through it (which is more pythonic).

But this code actually caches the value to the object.  So 1) it shouldn't be the same list object, and 2) the value on the object should actually be a list, even if the user passes a tuple, or the object's typing will be wrong.

File: e2e_tests/tests/test_sdk.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    exp.delete()
+    deleting_exp = exp
+
+    # Create another experiment and kill it.
+    exp = _make_live_experiment(client)
+    exp.kill()
+    assert exp.wait() == _client.ExperimentState.CANCELED
+
+    # Test remaining methods
+    exp.archive()
```
Comment (azhou-determined) [inline e2e_tests/tests/test_sdk.py:85]: do we need asserts for these status changes?

File: e2e_tests/tests/test_sdk.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    exp.pause()
+    with pytest.raises(ValueError, match="Make sure the experiment is active"):
+        # Wait throws an error if the experiment is paused by a user.
+        exp.wait(interval=0.1)
+
+    exp.activate()
+
+    exp.cancel()
+    assert exp.wait() == _client.ExperimentState.CANCELED
+    exp.wait()
```
Comment (azhou-determined) [inline e2e_tests/tests/test_sdk.py:71]: is this additional .wait() after the assert necessary?

**Review by azhou-determined — APPROVED** [review]
lgtm

File: e2e_tests/tests/test_sdk.py (hunk 4 above)
Comment (rb-determined-ai) [reply e2e_tests/tests/test_sdk.py:71]: nope, that's a carryover from some intermediate state of this code

File: e2e_tests/tests/test_sdk.py (hunk 3 above)
Comment (rb-determined-ai) [reply e2e_tests/tests/test_sdk.py:85]: I can add those.

**Pull request merged**